<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>562</width>
    <height>727</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>100</verstretch>
   </sizepolicy>
  </property>
  <property name="focusPolicy">
   <enum>Qt::StrongFocus</enum>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="ImageArea" name="widget" native="true">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>301</width>
      <height>251</height>
     </rect>
    </property>
   </widget>
   <widget class="ShowArea" name="showWidget" native="true">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>270</y>
      <width>541</width>
      <height>401</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
      <horstretch>100</horstretch>
      <verstretch>100</verstretch>
     </sizepolicy>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>2</width>
      <height>2</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_5"/>
   </widget>
   <widget class="QFontComboBox" name="fontComboBox">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>10</y>
      <width>213</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QSlider" name="fontsizeSlider">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>40</y>
      <width>19</width>
      <height>161</height>
     </rect>
    </property>
    <property name="maximum">
     <number>200</number>
    </property>
    <property name="value">
     <number>100</number>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="QSpinBox" name="sizeSpinBox">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>210</y>
      <width>91</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <number>255</number>
    </property>
    <property name="value">
     <number>100</number>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>240</y>
      <width>191</width>
      <height>20</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="text">
     <string>输入文字</string>
    </property>
    <property name="cursorPosition">
     <number>1</number>
    </property>
   </widget>
   <widget class="QPushButton" name="clearButton">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>170</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>清除图像</string>
    </property>
   </widget>
   <widget class="QPushButton" name="showButton">
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>170</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>显示预览</string>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QSlider" name="fontspaceSlider">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>70</y>
      <width>19</width>
      <height>131</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-100</number>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>210</y>
      <width>71</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>:大小/阈值</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>40</y>
      <width>54</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>文字间隔:</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="spaceSpinBox">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>40</y>
      <width>91</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <double>2.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.010000000000000</double>
    </property>
    <property name="value">
     <double>1.000000000000000</double>
    </property>
   </widget>
   <widget class="QCheckBox" name="encryptBox">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>80</y>
      <width>71</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>加密输出</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="fixedBox">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>100</y>
      <width>71</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>固定大小</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QCheckBox" name="reverseBox">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>120</y>
      <width>71</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>输出反转</string>
    </property>
   </widget>
   <zorder>layoutWidget</zorder>
   <zorder>widget</zorder>
   <zorder>showWidget</zorder>
   <zorder>fontComboBox</zorder>
   <zorder>fontsizeSlider</zorder>
   <zorder>sizeSpinBox</zorder>
   <zorder>lineEdit</zorder>
   <zorder>clearButton</zorder>
   <zorder>showButton</zorder>
   <zorder>fontspaceSlider</zorder>
   <zorder>label</zorder>
   <zorder>label_2</zorder>
   <zorder>spaceSpinBox</zorder>
   <zorder>encryptBox</zorder>
   <zorder>fixedBox</zorder>
   <zorder>reverseBox</zorder>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>562</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionLoad"/>
    <addaction name="actionSave"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menu_Edit">
    <property name="title">
     <string>&amp;Edit</string>
    </property>
    <addaction name="actionEditMatrix"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menu_Edit"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionOpen">
   <property name="text">
    <string>&amp;Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>E&amp;xit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>&amp;Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionEditMatrix">
   <property name="text">
    <string>&amp;Edit Matrix</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+E</string>
   </property>
  </action>
  <action name="actionLoad">
   <property name="text">
    <string>&amp;Load</string>
   </property>
   <property name="toolTip">
    <string>Load 3dwp file</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>ImageArea</class>
   <extends>QWidget</extends>
   <header>imagearea.h</header>
   <container>1</container>
   <slots>
    <signal>getFinished()</signal>
    <signal>getFinished(bool)</signal>
    <slot>getFontSize(int)</slot>
    <slot>setClearImage()</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>ShowArea</class>
   <extends>QWidget</extends>
   <header>showarea.h</header>
   <container>1</container>
   <slots>
    <signal>mapOver()</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>fontComboBox</sender>
   <signal>currentFontChanged(QFont)</signal>
   <receiver>MainWindow</receiver>
   <slot>fontChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>402</x>
     <y>55</y>
    </hint>
    <hint type="destinationlabel">
     <x>329</x>
     <y>81</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sizeSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>getThres(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>389</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>561</x>
     <y>288</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>clearButton</sender>
   <signal>released()</signal>
   <receiver>MainWindow</receiver>
   <slot>clearImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>478</x>
     <y>214</y>
    </hint>
    <hint type="destinationlabel">
     <x>561</x>
     <y>244</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>MainWindow</receiver>
   <slot>textChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>356</x>
     <y>284</y>
    </hint>
    <hint type="destinationlabel">
     <x>561</x>
     <y>352</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fontspaceSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>getSpaceSize(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>522</x>
     <y>173</y>
    </hint>
    <hint type="destinationlabel">
     <x>327</x>
     <y>168</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fontsizeSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>getFontSize(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>342</x>
     <y>139</y>
    </hint>
    <hint type="destinationlabel">
     <x>328</x>
     <y>212</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEdit</sender>
   <signal>selectionChanged()</signal>
   <receiver>lineEdit</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>493</x>
     <y>277</y>
    </hint>
    <hint type="destinationlabel">
     <x>523</x>
     <y>283</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>showButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>showPreview(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>413</x>
     <y>218</y>
    </hint>
    <hint type="destinationlabel">
     <x>430</x>
     <y>180</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>showWidget</sender>
   <signal>mapOver()</signal>
   <receiver>MainWindow</receiver>
   <slot>timerDone()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>14</x>
     <y>485</y>
    </hint>
    <hint type="destinationlabel">
     <x>4</x>
     <y>502</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fixedBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateAll()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>430</x>
     <y>142</y>
    </hint>
    <hint type="destinationlabel">
     <x>481</x>
     <y>141</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>widget</sender>
   <signal>getFinished(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateAll()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>115</x>
     <y>229</y>
    </hint>
    <hint type="destinationlabel">
     <x>153</x>
     <y>300</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>printInfos()</slot>
  <slot>getSize(int)</slot>
  <slot>saveImage()</slot>
  <slot>fontChanged()</slot>
  <slot>textChanged()</slot>
  <slot>getThres(int)</slot>
  <slot>clearImage()</slot>
  <slot>setTransFormed()</slot>
  <slot>updateStatus()</slot>
  <slot>getSpaceSize(int)</slot>
  <slot>getFontSize(int)</slot>
  <slot>showPreview(bool)</slot>
  <slot>timerDone()</slot>
  <slot>updateAll()</slot>
 </slots>
</ui>
